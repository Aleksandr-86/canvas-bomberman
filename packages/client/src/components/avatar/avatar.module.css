.profileImgSmall {
  width: 37px;
  height: 37px;
}

.profileImgMedium {
  width: 47px;
  height: 47px;
}

.profileImgLarge {
  width: 57px;
  height: 57px;
}

.profileImgGargantuan {
  width: 150px;
  height: 150px;
}

.profileImg {
  border-radius: 50%;
}

.clicable {
  cursor: pointer;
}
