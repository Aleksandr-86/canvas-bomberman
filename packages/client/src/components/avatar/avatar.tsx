import React from 'react'
import styles from './avatar.module.css'

interface Props {
  src: string
  size?: 'Small' | 'Medium' | 'Large' | 'Gargantuan'
  onAvatarClick?: () => void
  disabled?: boolean
}

export const Avatar: React.FC<Props> = props => {
  const { src, size = 'Medium', disabled = true, onAvatarClick } = props

  const handleClick = () => {
    if (disabled || !onAvatarClick) {
      return
    }

    onAvatarClick()
  }

  return (
    <div
      className={
        disabled
          ? styles.profilePicture
          : `${styles.profilePicture} ${styles.clickable}`
      }>
      <img
        src={src}
        className={`${styles.profileImg} ${styles['profileImg' + size]}`}
        alt="profile picture"
        onClick={handleClick}
      />
    </div>
  )
}
