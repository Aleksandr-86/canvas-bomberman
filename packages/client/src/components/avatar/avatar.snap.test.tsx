import * as renderer from 'react-test-renderer'
import { Avatar } from './avatar'

const mockSrc = `mockSrc.png`

describe(`<Avatar/> snapshot test cases`, () => {
  test(`<Avatar/> should render <img> with className profileImgMedium (default)`, () => {
    const tree = renderer.create(<Avatar src={mockSrc} />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  test(`<Avatar/> should render <img> with className profileImgSmall (size Small)`, () => {
    const tree = renderer.create(<Avatar src={mockSrc} size="Small" />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  test(`<Avatar/> should render <img> with className profileImgMedium (size Medium)`, () => {
    const tree = renderer
      .create(<Avatar src={mockSrc} size="Medium" />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  test(`<Avatar/> should render <img> with className profileImgLarge (size Large)`, () => {
    const tree = renderer.create(<Avatar src={mockSrc} size="Large" />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  test(`<Avatar/> should render <img> with className profileImgLarge (size Gargantuan)`, () => {
    const tree = renderer
      .create(<Avatar src={mockSrc} size="Gargantuan" />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
