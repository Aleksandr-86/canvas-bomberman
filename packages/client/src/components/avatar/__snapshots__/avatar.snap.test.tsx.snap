// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Avatar/> snapshot test cases <Avatar/> should render <img> with className profileImgLarge (size Gargantuan) 1`] = `
<div>
  <img
    alt="profile picture"
    className="profileImg profileImgGargantuan"
    onClick={[Function]}
    src="mockSrc.png"
  />
</div>
`;

exports[`<Avatar/> snapshot test cases <Avatar/> should render <img> with className profileImgLarge (size Large) 1`] = `
<div>
  <img
    alt="profile picture"
    className="profileImg profileImgLarge"
    onClick={[Function]}
    src="mockSrc.png"
  />
</div>
`;

exports[`<Avatar/> snapshot test cases <Avatar/> should render <img> with className profileImgMedium (default) 1`] = `
<div>
  <img
    alt="profile picture"
    className="profileImg profileImgMedium"
    onClick={[Function]}
    src="mockSrc.png"
  />
</div>
`;

exports[`<Avatar/> snapshot test cases <Avatar/> should render <img> with className profileImgMedium (size Medium) 1`] = `
<div>
  <img
    alt="profile picture"
    className="profileImg profileImgMedium"
    onClick={[Function]}
    src="mockSrc.png"
  />
</div>
`;

exports[`<Avatar/> snapshot test cases <Avatar/> should render <img> with className profileImgSmall (size Small) 1`] = `
<div>
  <img
    alt="profile picture"
    className="profileImg profileImgSmall"
    onClick={[Function]}
    src="mockSrc.png"
  />
</div>
`;
